### PR TITLE
[C++ for OpenCL] Added examples

### PR DESCRIPTION
**Pull request by AnastasiaStulova**
Added examples for construction and destruction of objects in local and constant address spaces.

### CODE REVIEW

File: cxx4opencl/address_spaces.txt [hunk 1]
```diff
@@ -343,10 +343,23 @@ their declaration. Developers are free to define a special member function
 that can initialize local address space objects after their declaration. Any
 default values provided for the initialization of members in a class
 declaration are ignored when creating the local address space objects.
-Classes that are instantiated with local address space objects can not
-have references as their members. The same restriction applies recursively
-to the members and base classes. Destructors of local address space objects
-are not invoked automatically. They can be called manually if required.
+Destructors of local address space objects are not invoked automatically.
```
Comment (AnastasiaStulova) [inline cxx4opencl/address_spaces.txt:346]: FYI this wording change is being reviewed in https://github.com/KhronosGroup/OpenCL-Docs/pull/509

File: cxx4opencl/address_spaces.txt [hunk 2]
```diff
@@ -358,6 +371,25 @@ Objects in `__constant` address space can be initialized using:
...
+[source,cpp]
+------------
+struct C1 {
+int m;
+};
+
+struct C2 {
+int m;
+C2(int init) __constant {};
+};
```
Comment (mantognini) [inline cxx4opencl/address_spaces.txt:383]:
```suggestion
struct C1 {
  int m;
};

struct C2 {
  int m;
  C2(int init) __constant {};
};
```

File: cxx4opencl/address_spaces.txt [hunk 3]
```diff
@@ -358,6 +371,25 @@ Objects in `__constant` address space can be initialized using:
...
+
+kernel void k()
+{
```
Comment (mantognini) [inline cxx4opencl/address_spaces.txt:386]:
```suggestion
kernel void k() {
```

**Review by mantognini — COMMENTED** [review]
Some cosmetics suggestions.

**Comment by AnastasiaStulova** [comment]
Rebased against master

**Review by mantognini — APPROVED** [review]
LGTM, thanks for the update.

File: cxx4opencl/address_spaces.txt [hunk 4]
```diff
@@ -354,6 +354,20 @@ is required for classes with data members that are references because their valu
...
+------------
+class C {
+  int m;
+};
+
+kernel void foo() {
+  __local C locobj{}; // error: local address space objects can't be initialized
+  __local C locobj; // uninitialised object
+  locobj = {}; // calling copy assignment operator is allowed
+  locobj.~C();
```
Comment (svenvh) [inline cxx4opencl/address_spaces.txt:367]:
```suggestion
  locobj.~C(); // local address space object destructors are not invoked automatically.
```

For consistency with the lines above.

**Review by svenvh — APPROVED** [review]

**Pull request merged**